javascript/ql/src/Security/CWE-327/BadRandomness.ql

```diff
@@ -0,0 +1,109 @@
+/**
+ * @name Creating biased random from unbiased source.
+ * @description Some mathematical operations on random numbers can cause bias in
+ *              the results and compromise security.
+ * @kind problem
+ * @problem.severity warning
+ * @precision high
+ * @id js/biased-cryptographic-random
+ * @tags security
+ *       external/cwe/cwe-327
+ */
+
+import javascript
+private import semmle.javascript.dataflow.InferredTypes
+
+/**
+ * Gets a source of cryptographically random numbers.
+ */
+DataFlow::SourceNode randomSource() {
+  result = DataFlow::moduleMember("crypto", ["randomBytes", "randomFillSync"]).getACall()
```

Comment:
Huh. I thought there would be more relevant members of `crypto` here.
Related: have you considered some of the most popular crypto packages on npm? (they are probably just useful wrappers around `crypto`)

Reply:
Most of `crypto` is for encryption/hashing. 

Same with crypto packages on npm. They are generally designed for encryption/hashing, and generally don't have an API for secure random number generation.  

[secure-random](https://www.npmjs.com/package/secure-random) was the best I could find, and that has ~20.000 weekly downloads.  (I've added support for that).

Reply:
There are libraries for generating random strings. (Including [the library from the CVE](https://hapi.dev/module/cryptiles/api/?v=5.1.0)).   
But I don't think random strings is a source for this query.